File: font_end/lib/models/payment_card.dart
class PaymentCard {
  final String id;
  final String cardHolderName;
  final String cardNumber;
  final String expiryDate;
  final String brand;
  final bool isDefault;

  PaymentCard({
    required this.id,
    required this.cardHolderName,
    required this.cardNumber,
    required this.expiryDate,
    required this.brand,
    required this.isDefault,
  });

  factory PaymentCard.fromJson(Map<String, dynamic> json) {
    return PaymentCard(
      id: json['id'],
      cardHolderName: json['cardHolderName'] ?? '',
      cardNumber: json['cardNumber'] ?? '',
      expiryDate: json['expiryDate'] ?? '',
      brand: json['brand'] ?? '',
      isDefault: json['isDefault'] ?? false,
    );
  }
}

File: back_end/src/main/java/com/nguyenquocbao/back_end/entity/PaymentCard.java
package com.nguyenquocbao.back_end.entity;

import jakarta.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import java.util.UUID;

@Entity
@Table(name = "payment_cards")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentCard {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    private String cardHolderName;
    private String cardNumber;
    private String expiryDate;
    private String brand;
    private Boolean isDefault;
}
